apps/gradient_autoscheduler/GradientAutoscheduler.cpp

```diff
@@ -179,6 +232,18 @@ void parallelize_vars_and_rvars_gpu(
                             << fused_rvar << ")\n";
         }
     }
+    // CUDA places rather strict restriction on the second dimension of the GPU blocks (usually 65536),
+    // so we want to split it if it is too large
+    int rdomain_size = 1;
+    for (int b : rvar_bounds) {
+        rdomain_size *= b;
+    }
+    std::string fused_rvar2;
+    if (rdomain_size >= 65536) {
+        RVar r;
```

Comment:
Suggestion: give basename like "cuda_gpu_block_split" or something, just to make future debugging easier.

Reply:
Given!